cumulus_etl/nlp/openai.py

```diff
@@ -10,37 +10,62 @@
 from cumulus_etl import errors
 
 
-class OpenAIModel(abc.ABC):
```

Comment:
I'm changing away from an abstract class that is inherited by specific-use models like AzureModel and LocalModel. And switching that into a class that looks for all supported methods (azure, bedrock, local) and finds whichever mode is configured. This better supports stuff like llama4, which can do any of the three modes depending on config.